スタンダード第30回演習課題です。repositoryのテストを作りました。

Shintaro0806 opened the pull request:
以前、質問投稿で、data.sqlとapplication.propertiesとschema.sqlをあげているため、課題の範囲すべてのデータではないかもしれませんが、よろしくお願いします。

Commit:
スタンダード第30回演習課題です。repositoryのテストを作りました。
以前、質問投稿で、data.sqlとapplication.propertiesとschema.sqlをあげているため、課題の範囲すべてのデータではないかもしれませんが、よろしくお願いします。

KeitaUenishi commented on src/test/java/raisetech/Re/Student/Management/repository/StudentRepositoryTest.java:
```diff
@@ -18,9 +23,59 @@ class StudentRepositoryTest {
+  void 受講生をIDで検索ができること() {
+    Student actual = sut.searchStudent("1");
+
+    assertThat(actual).isNotNull();
+    assertThat(actual.getId()).isEqualTo("1");
+    assertThat(actual.getName()).isEqualTo("山田太郎");
+    assertThat(actual.getKanaName()).isEqualTo("ヤマダタロウ");
+    assertThat(actual.getNickname()).isEqualTo("タロ");
+    assertThat(actual.getEmail()).isEqualTo("taro@example.com");
+    assertThat(actual.getArea()).isEqualTo("東京");
+    assertThat(actual.getAge()).isEqualTo(25);
+    assertThat(actual.getSex()).isEqualTo("男性");
```
equalsとhashCodeを実装すると、インスタンスそのものを比較できるようになるのでそちらで試してみていただければと！
https://qiita.com/yachinco/items/b1a3602d6aa261f5f7c4

Commit:
スタンダード第30回演習課題を修正しました。equalsとhashCodeを実装してみたのですが、これでよいでしょうか。

KeitaUenishi commented on src/test/java/raisetech/Re/Student/Management/repository/StudentRepositoryTest.java:
```diff
@@ -29,17 +29,13 @@ class StudentRepositoryTest {
-    assertThat(actual.getArea()).isEqualTo("東京");
-    assertThat(actual.getAge()).isEqualTo(25);
-    assertThat(actual.getSex()).isEqualTo("男性");
+    assertThat(actual).usingRecursiveComparison().isEqualTo(expected);
```
OKです！ ここだけではなく、他の値ごとに検証している箇所も同じように修正してしまいましょう！

Commit:
スタンダード第30回演習課題を修正しました。
受講生のコース情報をIDで検索ができること、をインスタンスで比較できるようにしました。他はsizeで比較していたり、一か所を更新して、それを比較しているためそのままにしています。

KeitaUenishi commented on src/main/java/raisetech/Re/Student/Management/data/StudentCourse.java:
```diff
@@ -25,4 +26,26 @@ public class StudentCourse {
+    this.courseName = courseName;
+    this.courseStartAt = courseStartAt;
+    this.courseEndAt = courseEndAt;
+  }
```
こちらですが、 `@NoArgsConstructor` を追加されているように、`@AllArgsConstructor` を追加することでlombokで自動生成してくれるようになります！

Commit:
スタンダード第30回演習課題を修正しました。
受講生のコース情報の登録が行えること、をコンストラクタを用いて見やすくし、また、StudentCourseにAllArgConstructorをつけて見やすくしました。
